src/databricks/labs/ucx/aws/credentials.py

```diff
@@ -230,3 +230,30 @@ def run(self, prompts: Prompts, *, single_role=False, role_name="UC_ROLE", polic
 
         self._resource_permissions.create_uc_roles(iam_list)
         self._resource_permissions.save_uc_compatible_roles()
+
+    def delete_uc_roles(self, prompts: Prompts):
+        uc_roles = self._resource_permissions.load_uc_compatible_roles()
+        if len(uc_roles) == 0:
+            self._resource_permissions.save_uc_compatible_roles()
+            uc_roles = self._resource_permissions.load_uc_compatible_roles()
+        storage_credentials = self._ws.storage_credentials.list()
+
+        uc_role_dict = {role.role_arn: role.role_name for role in uc_roles}
+        selected_roles = prompts.multiple_choice_from_dict("Select the list of roles created by UCX", uc_role_dict)
```

Comment:
We can not determine this ourselves? Maybe with tags?

Reply:
the create-missing-credential cmd takes a role name as cmd and it could be single or multiple and we create the UC roles based on that. We don't store the role_name passed so the name could be anything and we don't stage it, hence the option to give users to select from the choice